test/auth.test.js

```diff
@@ -94,6 +102,127 @@ describe('Login', () => {
   });
 });
 
+describe('Validation', () => {
+  it('should return false if the token is wrong', (done) => {
+    const valid = isTokenValid('wrongtoken', data);
+    valid.should.be.a('boolean').equal(false);
+    done();
+  });
+});
+
+describe('Password reset', () => {
+  describe('Email sending', () => {
+    it('it should send an email to the user', (done) => {
+      chai
+        .request(server)
+        .post('/api/reset')
+        .send({ email: data.email })
+        .end((err, res) => {
+          res.should.have.status(200);
+          done();
+        });
+    });
+    it('it should no send an email when an invalid email addres is provided', (done) => {
+      chai
+        .request(server)
+        .post('/api/reset')
+        .send({ email: 'email.email' })
+        .end((err, res) => {
+          res.should.have.status(400);
+          done();
+        });
+    });
+    it('it should no send an email when an unknown email addres is provided', (done) => {
+      chai
+        .request(server)
+        .post('/api/reset')
+        .send({ email: 'email@email.com' })
+        .end((err, res) => {
+          res.should.have.status(404);
+          done();
+        });
+    });
+  });
+  describe('Link verification', () => {
+    const token = generateLink(' ', { email: data.email }).split(' /')[1];
+    it('it should verify the link sent to the user', (done) => {
+      chai
+        .request(server)
+        .get(`/api/reset/${token}`)
+        .end((err, res) => {
+          res.should.have.status(200);
+          userEmail = res.body.data.email;
```

Comment:
are you using this `userEmail` variable?

Reply:
Yes. I'm using it